src/main/java/com/meilisearch/sdk/Client.java

```diff
@@ -41,20 +44,19 @@ public Index createIndex(String uid) throws Exception {
      *
      * @param uid Unique identifier for the index to create
      * @param primaryKey The primary key of the documents in that index
-     * @return Meilisearch API response
+     * @return Meilisearch API response as Task
      * @throws Exception if an error occurs
      */
-    public Index createIndex(String uid, String primaryKey) throws Exception {
-        Index index = gson.fromJson(this.indexesHandler.create(uid, primaryKey), Index.class);
-        index.setConfig(this.config);
```

Comment:
Sorry for the dumb question, but what did this method do before? Is its absence will be missed since we are not responding with an index now?

Reply:
This method creates an index on `Meilisearch`. It will still create an index but in an asynchronous way and return a task.
You have different way to [create an index](https://docs.meilisearch.com/learn/core_concepts/indexes.html#index-creation) with the `createIndex()` method:
```
Task task = client.createIndex("movies");
... wait ...
Index index = client.getindex("movies");
```
Or with the implicit index creation by passing by `addDocument()` method:
```
Index index = client.index("movies");
Task task = index.addDocuments(<document_to_add>);
```
You can check the details of PR #311